### PR TITLE
Doc: Updates to resiliency page for 1.5.0/1

### PR DESCRIPTION
**Pull request by clintongormley**
@bleskes @s1monw @kimchy Please could you take a look and let me know if I've missed anything

### CODE REVIEW

File: docs/resiliency/index.asciidoc [hunk 1]
```diff
@@ -156,24 +159,60 @@ Almost all files in Elasticsearch now have checksums which are validated before
...
+Now, shards from versions before v1.4.0 are copied over in full and recovery from versions
+before v1.3.2 are disabled entirely. See {GIT}9925[#9925] (STATUS: DONE, Fixed in v1.5.0)
+
+
+[float]
+=== Upgrade 3.x segments on engine startup (STATUS: DONE, v1.5.0)
+
+Upgrading old 3.x segments can be error prone and can result in corruption when merges are being
+run concurrently. Instead, Elasticsearch will now upgrade 3.x segments in place before the
+engine starts.  See {GIT}9899[#9899] (STATUS; DONE, fixed in v1.5.0)
```
Comment (kimchy) [inline docs/resiliency/index.asciidoc]: I think this is misleading, its not upgrading the segments, it just writes a new commit point using the new version

**Comment by kimchy** [comment]
left one comment, other than that, LGTM

File: docs/resiliency/index.asciidoc [hunk 2]
```diff
@@ -156,24 +159,60 @@ Almost all files in Elasticsearch now have checksums which are validated before
...
+
+Commonly used filters are cached in Elasticsearch. That cache is limited in size (10% of node's memory by default) and is being evicted based on a least recently used policy. The amount of memory used by the cache depends on two primary components - the values it stores and the keys associated with them. Calculating the memory footprint of the values is easy enough but the keys accounting is trickier to achieve as they are, by default, raw Lucene objects. This is largely not a problem as the keys are dominated by the values. However, recent optimizations in Lucene have changed the balance causing the filter cache to grow beyond it's size.
+
+While we are working on a longer term solution ({GIT}9176[#9176]), we introduced a minimum weight of 1k for each cache entry. This puts an effective limit on the number of entries in the cache. See {GIT}8304[#8304] (STATUS: DONE, fixed in v1.4.0)
+
+== Completed
 
 [float]
-=== Prevent setting minimum_master_nodes to more than the current node count (STATUS: ONGOING)
+=== Ensure cluster state ID is incremental (STATUS: DONE, v1.5.1)
```
Comment (bleskes) [inline docs/resiliency/index.asciidoc]: This is about local shard state files, not cluster state.

File: docs/resiliency/index.asciidoc [hunk 3]
```diff
@@ -156,24 +159,60 @@ Almost all files in Elasticsearch now have checksums which are validated before
...
+
+It is possible that the current state ID can be reset or even go backwards
+when nodes or the cluster are restarted, or nodes with legacy data join
```
Comment (bleskes) [inline docs/resiliency/index.asciidoc]: We never full researched how this could have happened. As far as I can tell, this can happen on a full cluster restart (or when all copies of the shard are gone) which happens while the nodes are writing these state file. On top of that the new shards have to be assigned before all shard copies are present back in the cluster. I would reword this sentence to something like ".. go backwards in very extreme cases during full cluster restart".

**Comment by clintongormley** [comment]
Changes addressed - thanks.
